hudi-common/src/main/java/org/apache/hudi/metadata/FileSystemBackedTableMetadata.java

```diff
@@ -163,8 +165,13 @@ private List<String> getPartitionPathWithPathPrefixUsingFilterExpression(String
           "Listing all partitions with prefix " + relativePathPrefix);
       // Need to use serializable file status here, see HUDI-5936
       List<StoragePathInfo> dirToFileListing = engineContext.flatMap(pathsToList, path -> {
-        HoodieStorage storage = HoodieStorageUtils.getStorage(path, storageConf);
-        return storage.listDirectEntries(path).stream();
+        try {
+          HoodieStorage storage = HoodieStorageUtils.getStorage(path, storageConf);
+          return storage.listDirectEntries(path).stream();
+        } catch (FileNotFoundException e) {
+          // The partition may have been cleaned.
+          return Stream.empty();
+        }
```

Comment:
If we enable Partition TTL, partition may be cleaned by Task Manager and there will throw a FileNotFoundException.